src/Compilers/CSharp/Portable/Symbols/TypeSymbolExtensions.cs

```diff
@@ -296,6 +296,7 @@ public static bool IsValidExtensionParameterType(this TypeSymbol type)
                 case TypeKind.Pointer:
                 case TypeKind.Dynamic:
                 case TypeKind.FunctionPointer:
+                case TypeKind.Extension:
```

Comment:
I'm honestly not sure why this would be the case? This feels like it should just work.